lib/r10k/action/deploy/module.rb

```diff
@@ -77,15 +77,12 @@ def visit_puppetfile(puppetfile)
         end
 
         def visit_module(mod)
-          requested_mods = @settings.dig(:overrides, :modules, :requested_modules)
-          if requested_mods.include?(mod.name)
-            mod.sync
-            if mod.environment && @settings.dig(:overrides, :environments, :generate_types)
-              logger.debug("Generating puppet types for environment '#{mod.environment.dirname}'...")
-              mod.environment.generate_types!
-            end
-          else
-            logger.debug1(_("Only updating modules %{modules}, skipping module %{mod_name}") % {modules: requested_mods.inspect, mod_name: mod.name})
+          mod.sync
```

Comment:
I'm going through this commit-by-commit, so maybe I'll come around to this as I move forward, but... at first blush, I feel like this change (1901284b05164f920c882a3281f062ecb6dafac2) actually makes the code _more_ confusing, rather than less. Admittedly, I've been working with r10k for a little while, so may be looking a little more favorably on visitor stuff than I would have when I first started trying to figure it out.

In general, I don't like the idea of putting decision-making code into module objects. I am in favor of keeping module objects dumb, being given only simple instructions and doing what they're told. I think we should be able to find a way to simplify the visitor stuff without losing that simplicity for module objects.